src/software/gui/drawing/navigator.cpp

```diff
@@ -22,11 +22,14 @@ AIDrawFunction drawNavigator(std::shared_ptr<Navigator> navigator)
 
         for (const auto& path : planned_paths)
         {
-            for (unsigned i = 0; i < path.size() - 1; i++)
+            if (path.size())
```

Comment:
Why is the extra `if` necessary? Isn't this the same behavior as just the loop?

Reply:
The new API now has us returning empty vector for no path. This handles that here

Reply:
Yea but if the vector is empty the loop wouldn't run anyway?

Reply:
We want to check that path.size() is greater than 0

Reply:
It turns out that without this change, it segfaults

Reply:
Just so we're being super explicit, could we make this:
``` suggestion
            if (path.size() > 0)
```